applications/GeoMechanicsApplication/custom_processes/apply_c_phi_reduction_process.cpp

```diff
@@ -74,17 +83,23 @@ void ApplyCPhiReductionProcess::ExecuteFinalize()
 
 int ApplyCPhiReductionProcess::Check()
 {
-    KRATOS_ERROR_IF(mrModelPart.Elements().empty())
-        << "ApplyCPhiReductionProces has no elements in modelpart " << mrModelPart.Name() << std::endl;
-    return 0;
+    for (const auto& r_model_part : mrModelParts) {
+        if (!r_model_part.get().Elements().empty()) return 0;
+    }
+    for (const auto& r_model_part : mrModelParts) {
+        KRATOS_INFO("ApplyCPhiReductionProcess")
+            << " modelpart " << r_model_part.get().Name() << " has no elements." << std::endl;
+    }
+    KRATOS_ERROR << "ApplyCPhiReductionProcess needs at least one element." << std::endl;
```

Comment:
I believe there should be one check only, not two. In addition, we can express the check slightly clearer by using an STL algorithm:
```suggestion
    KRATOS_ERROR_IF(std::ranges::all_of(mrModelParts, [](const auto& r_model_part_ref){ return r_model_part_ref.get().Elements().empty(); })) << "None of the provided model parts contains at least one element. A c-phi reduction analysis requires at least one element.\n";
    
    return 0;
```

Reply:
That's real C++